common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -58,8 +58,15 @@ class HTTPClient(
     private fun getInputStream(connection: HttpURLConnection): InputStream? {
         return try {
             connection.inputStream
-        } catch (e: IOException) {
-            connection.errorStream
+        } catch(e: Exception) {
```

Comment:
At least in Java you can do `catch(e: IOException | IllegalArgumentException)`. I wonder if that works here too?

Reply:
Seven. Years. Old.
https://youtrack.jetbrains.com/issue/KT-7128

Reply:
Haha wow

Reply:
Detekt complains about catching a too generic Exception, which is fine to ignore. Alternatively, you could do the multiple `catch` version and create a function (I think):

```
try {

catch (e: IllegalArgumentException) {
  handleException(e)
catch (e: IOException) {
  handleException(e)
}
```